ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java

```diff
@@ -124,14 +125,11 @@ public PeerDiscoveryAgent(
       LOG.info("Starting peer discovery agent on host={}, port={}", host, port);
 
       // override advertised host if we detect an external IP address via NAT manager
-      final String advertisedAddress;
-      if (natService.isPresent()) {
-        advertisedAddress =
-            natService.get().queryExternalIPAddress().orElseGet(config::getAdvertisedHost);
-      } else {
-        advertisedAddress = config.getAdvertisedHost();
-      }
-
+      final AtomicReference<String> advertisedAddress = new AtomicReference<>();
```

Comment:
Why does this need to be an AtomicReference?

Reply:
you are right it is useless. I just simplified this part